python/ray/data/_internal/logical/interfaces.py

```diff
@@ -51,11 +54,57 @@ def __init__(self, name: str, input_dependencies: List["LogicalOperator"]):
             assert isinstance(x, LogicalOperator), x
 
 
+class Plan:
+    """Abstract class for logical/physical execution plans.
+
+    This plan should hold an operator representing the plan DAG and any auxiliary data
+    that's useful for plan optimization or execution.
+    """
+
+    @property
+    def dag(self) -> Operator:
+        raise NotImplementedError
+
+
+class LogicalPlan(Plan):
+    """The plan with a DAG of logical operators."""
+
+    def __init__(self, dag: LogicalOperator):
+        self._dag = dag
+
+    @property
+    def dag(self) -> LogicalOperator:
+        """Get the DAG of logical operators."""
+        return self._dag
+
+
+class PhysicalPlan(Plan):
```

Comment:
The file path is "logical/interfaces.py" but we are now introducing physical pieces. Shall we name the path as optimizer? It'll be consistent of the 3 components of query processing (planner, optimizer, execution).

Reply:
I am fine for the renaming if others have no objection. But wanted to make sure we are addressing renaming in a separate PR, for easier review.

Reply:
There is already a "planner/" directory, why the "Plan" and related are not belong to "planner/"?

Reply:
I guess it's fine to put `Plan/LogicalPlan/PhysicalPlan` here, as `Rule` depends on `Plan`, and `Optimizer` depends on `Rule`. In the future, we may generalize some graph traversal logic into `Plan`.